lock-api-objects/src/main/java/com/palantir/lock/LockRequest.java

```diff
@@ -200,15 +200,14 @@ public boolean equals(@Nullable Object obj) {
                 && lockTimeout.equals(other.lockTimeout)
                 && (lockGroupBehavior == other.lockGroupBehavior)
                 && (blockingMode == other.blockingMode)
-                && Objects.equal(blockingDuration, other.blockingDuration)
-                && Objects.equal(versionId, other.versionId);
+                && Objects.equals(blockingDuration, other.blockingDuration)
+                && Objects.equals(versionId, other.versionId);
     }
 
     @Override
     public int hashCode() {
         if (hashCode == null) {
-            hashCode = Objects.hashCode(
-                    lockMap, lockTimeout, lockGroupBehavior, blockingMode, blockingDuration, versionId);
+            hashCode = Objects.hash(lockMap, lockTimeout, lockGroupBehavior, blockingMode, blockingDuration, versionId);
```

Comment:
I was a bit worried about this one because it could lead to fun times (if you deserialize something from the old version that has a different way of calculating the hash code and shove it into a set you have fun times), but think it's safe given that you can only create this through the serialization proxy, meaning that hash codes are always computed with the current JVM. I'd appreciate a double-check though.

Reply:
I don’t think we support java serialization across versions

Reply:
That said, the hashCode field on serializable objects should be transient, and there's no reason for it to be volatile.

Reply:
Not sure why we're doing all this boxing/unboxing either. We should use an `int` with a sentinel value (zero) for the uninitialized case. Some objects may be unlucky, however the hashcode is relatively inexpensive to recompute.